r/R/dataset-scan.R

```diff
@@ -214,13 +228,7 @@ map_batches <- function(X, FUN, ..., .data.frame = TRUE) {
     res <- res[seq_len(i)]
   }
 
-  if (.data.frame & inherits(res[[1]], "arrow_dplyr_query")) {
-    res <- dplyr::bind_rows(map(res, dplyr::collect))
-  } else if (.data.frame) {
-    res <- dplyr::bind_rows(map(res, as.data.frame))
-  }
-
-  res
+  as_adq(RecordBatchReader$create(batches = res))
```

Comment:
Question: Why return an ADQ instead of a straight RBR? RBR is a documented type, while ADQ is undocumented, and I think you can still use a RBR in a dplyr pipeline, right?

Reply:
I was thinking back to the discussion that ARROW-15489 refers to, where RBR didn't print usefully so it would be weird to get one back. But now that that issue has been done, you're right, RBR makes sense (it prints about the same as ADQ now). I can switch this one and also switch the duckdb to_arrow() one as well.

Reply:
Done in https://github.com/apache/arrow/pull/13170/commits/e12c633327439fef6857ea8c6b08a18af2301c96